src/outputs/parthenon_hdf5.cpp

```diff
@@ -15,17 +15,193 @@
 // the public, perform publicly and display publicly, and to permit others to do so.
 //========================================================================================
 
+// options for building
+#include "config.hpp"
+
 // Only proceed if HDF5 output enabled
+#ifdef HDF5OUTPUT
 
+#include <algorithm>
 #include <memory>
+#include <set>
+#include <type_traits>
+#include <unordered_map>
 
+#include "interface/meshblock_data_iterator.hpp"
+#include "interface/metadata.hpp"
+#include "mesh/mesh.hpp"
+#include "mesh/meshblock.hpp"
+#include "outputs/outputs.hpp"
 #include "outputs/parthenon_hdf5.hpp"
 
-#include "mesh/meshblock.hpp"
+namespace parthenon {
+namespace HDF5 {
+
+// template specializations for std::string
+template <>
+void HDF5WriteAttribute(const std::string &name, const std::vector<std::string> &values,
+                        hid_t location) {
+  std::vector<const char *> char_ptrs(values.size());
+  for (size_t i = 0; i < values.size(); ++i) {
+    char_ptrs[i] = values[i].c_str();
+  }
```

Comment:
I would use a range based for loop, secondly I would call reserve instead of initializing, then I would use emplace.

Reply:
Normally, I agree that it's better to use range based for loops. However, in this case, we need the index so we can assign the relevant entry in `char_ptrs`. You're right that we could use `emplace_back`. However, since a `const char *` is default-constructible, we don't have to and in this case, using `reserve` and `emplace_back` seems too convoluted and would even add an extra line. We could use `std::transform`, but I think the plain index based for loop is the most readable and easy to understand. And there are no performance reasons to not use it.